client/src/index.ts

```diff
@@ -0,0 +1,48 @@
+const notesGrid: HTMLElement|null = document.querySelector('tbody');
```

Comment:
The whole file is just a mess of different functions without any structure. 

First of all there is should be one entry point, smth like 

```typescript
class Main {
    public static main(): void {
        // your logic
    }
}

Main.main();
```


Then in the main static method you can create a function / class to do smth, usually this should be placed in separate files, but since we don't have a lot of logic, we can put it right here, but at least it should have some structure.

You can use class / functions or simple objects for this (class is preferred)
For example:
```typescript
class TableCreator {
    constructor(private readonly tableSelector: string, private readonly serverUrl: string) {

    }

    private async getDataFromServer(): Promise<Note[]> {
        return fetch(this.serverUrl)
            .then(res => res.json());
    }

    private createNoteElement(note: Note): Element {}

    async renderTable(): void {
        const tableElement = document.querySelector(this.tableSelector);
        if (!tableElement)
            throw new Error('INT: There is no table element at the page');

        const data = await this.getDataFromServer();
        for (let i = 0; i < data.length; ++i) {
            tableElement.appendChild(this.createNoteElement(data[i]));
        }
    }
}


new TableCreator('tbody', 'http://localhost:5000/').renderTable();
```

Even the example I've written contains a lot of drawback, but at least it shows point of view.